### PR TITLE
Bump for ruby 2.7 keyword args

### PR DESCRIPTION
**Pull request by kklimuk**

### CODE REVIEW

File: .ruby-version [hunk 1]
```diff
@@ -1 +1 @@
-2.4.0
\ No newline at end of file
+2.7.0
```
Comment (krnjn) [inline .ruby-version]: Not a huge deal -- but I think Ruby 2.7.1 is out (with some CVE fixes so may be worth bumping to that version specifically).

File: lib/ruby_memoized/memoizer.rb [hunk 2]
```diff
@@ -7,13 +7,13 @@ def initialize(context, method)
       @method = method
     end
 
-    def call(*args, &block)
-      return cache[[args, block]] if cache.has_key?([args, block])
-      cache[[args, block]] = context.send(method, *args, &block)
+    def call(*args, **kwargs, &block)
```
Comment (krnjn) [inline lib/ruby_memoized/memoizer.rb:10]: I think this works for updating the cache, but does this defining of the methods that you do [here](https://github.com/kklimuk/ruby_memoized/blob/master/lib/ruby_memoized.rb#L37) also need to receive the same type of update to cover the kwargs case?

File: spec/ruby_memoized/memoizer_spec.rb [hunk 3]
```diff
@@ -15,41 +15,46 @@
 
   let(:method) { :american_exceptionalism }
 
-  subject { memoizer.call(*args, &block) }
+  subject { memoizer.call(*args, **kwargs, &block) }
```
Comment (krnjn) [inline spec/ruby_memoized/memoizer_spec.rb:18]: On line 12 above (in this file), do you have to do some similar double splatting on the class definition you've added? https://github.com/kklimuk/ruby_memoized/pull/2/files#diff-83353dc8cb2fcf868660eccb407f798fR12 (just comparing against what has been changed, but not sure).

**Review by krnjn — COMMENTED** [review]
Thanks for your quick work on this! Left some comments about a few other methods that I noticed were not changed in the same way the main `call` method was. Let me know if they make sense / warrant an update as well.

**Comment by krnjn** [comment]
@kklimuk I left some comments on Friday. LMK if they were confusing / happy to help clarify!

**Review by krnjn — CHANGES_REQUESTED** [review]
Just adding a request changes in case the comments weren't posting!

**Comment by kklimuk** [comment]
Hey @krnjn, sorry about that, was away for a few days. Taking a look now.

File: spec/ruby_memoized/memoizer_spec.rb (hunk 3 above)
Comment (kklimuk) [reply spec/ruby_memoized/memoizer_spec.rb:18]: yep, this happens in `memoizer.rb`

File: lib/ruby_memoized/memoizer.rb (hunk 2 above)
Comment (kklimuk) [reply lib/ruby_memoized/memoizer.rb:10]: 🤦 great point! fixed

File: spec/ruby_memoized/memoizer_spec.rb (hunk 3 above)
Comment (krnjn) [reply spec/ruby_memoized/memoizer_spec.rb:18]: Got it. I guess what I was saying was do you need to change `define_method(method_name) { |*_, &_| output_value }` to `define_method(method_name) { |*_, **_, &_| output_value }` or something like that. 

Not sure, but just another place where it feels like there's a reference to the old style `*args, &block` that has now been changed to `*args, **kwargs, &block` everywhere (obviously with the caveat that it's underscore non-used methods).

**Review by krnjn — APPROVED** [review]
Aside from the one minor comment, this LGTM!

File: spec/ruby_memoized/memoizer_spec.rb (hunk 3 above)
Comment (kklimuk) [reply spec/ruby_memoized/memoizer_spec.rb:18]: ah, i see your point

Comment (kklimuk) [reply spec/ruby_memoized/memoizer_spec.rb:18]: fixed

**Comment by krnjn** [comment]
Thanks for the quick fix @kklimuk!

**Comment by egiurleo** [comment]
Hi, @kklimuk! If you have time, could you make a release with this change? It would be great to be able to upgrade our version of ruby_memoized :) Thanks!

**Pull request merged**